src/ggml.c

```diff
@@ -12155,6 +12199,67 @@ static void ggml_compute_forward_upscale(
     }
 }
 
+// ggml_compute_forward_pad
+
+static void ggml_compute_forward_pad_f32(
+    const struct ggml_compute_params * params,
+    const struct ggml_tensor * src0,
+    struct ggml_tensor * dst) {
+
+    if (params->type == GGML_TASK_INIT || params->type == GGML_TASK_FINALIZE) {
+        return;
+    }
+
+    GGML_ASSERT(src0->nb[0] == sizeof(float));
+    GGML_ASSERT(dst->nb[0] == sizeof(float));
+
+    const int ith = params->ith;
+    const int nth = params->nth;
+
+    GGML_TENSOR_UNARY_OP_LOCALS
+
+    const int padding_factor = dst->op_params[0];
+
+    float* src0_ptr = (float*)src0->data;
+    float* dst_ptr = (float*)dst->data;
+
+    const so2 = ne00 * ne01;
+    const so3 = ne00 * ne01 * ne02;
+    const do2 = ne0 * ne1;
+    const do3 = ne0 * ne1 * ne2;
```

Comment:
This should use the tensor `nb` instead. You can divide `nb` by `sizeof float` to obtain element strides as used here, instead of a number of bytes.

As it is, this requires the tensor to be contiguous in memory, but after changing it to use `nb` it will only require that the rows are contiguous. It should check for that by doing an assert `nb0 == sizeof(float)` and `nb00 == sizeof(float)`.